### DISCUSSION TITLE
Nested stores with SSR

### DISCUSSION THREAD

**Discussion by m00nk** [vuejs/pinia#1315] [Help and Questions] [answered]
I have several stores and some of them use some others. How do I create the stores inside the other stores for good working in SSR? I read I need to send the pinia object (root store object) into the store constructor when I create the stores for SSR like this:

```js
import {createPinia} from "pinia";

import MyStore from 'my-store';

app.use(pinia);

const myStore = MyStore(pinia);
```

But do I need to do the same thing inside the store code to have an ability to use the other stores? Example:

```js
// file my-store.js

import {defineStore} from 'pinia';
import MyOtherStore from 'my-other-store';

export default defineStore('my-store', {
    // ...
    actions: {
         someAction(){
              const other = MyOtherStore(/* 1 */);
         }
    }
}
);
```

Do I need to send the pinia object in (1) for SSR working right? If so, how can I get it?

**Comment by posva** [comment, marked as answer]
No, you don’t need it
